flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/LocalStreamEnvironment.java

```diff
@@ -91,7 +91,11 @@ public JobExecutionResult execute(String jobName) throws Exception {
 		configuration.addAll(jobGraph.getJobConfiguration());
 
 		configuration.setLong(ConfigConstants.TASK_MANAGER_MEMORY_SIZE_KEY, -1L);
-		configuration.setInteger(ConfigConstants.TASK_MANAGER_NUM_TASK_SLOTS, getParallelism());
+
+		int parallelism = getParallelism() == defaultLocalParallelism ?
+				defaultLocalParallelism : jobGraph.getMaximumParallelism();
```

Comment:
`defaultLocalParallelism` seems to be the number of available processors. I think this is very unintuitive that the default parallelism is only taken if it equals the number of processors.

This means in case of an operator with a higher dop than the #cores, the program will only fail if the user sets the default parallelism to #cores.